src/test/java/io/vantiq/client/intg/VantiqIntegrationTest.java

```diff
@@ -431,11 +431,15 @@ public void testSubscribeReliableTopic() throws Exception {
         vantiq.subscribe(Vantiq.SystemResources.TOPICS.value(), "/test/topic", null, callback, params);
         callback.waitForCompletion();
         LinkedTreeMap msg = (LinkedTreeMap) callback.getMessage().getBody();
-        String ackId = msg.get("subscriptionName").toString();
-        String requestId = msg.get("requestId").toString();
-        callback.reset();
-        callback.waitForCompletion();
-        assertThat("Connected", callback.isConnected(), is(true));
+        assertThat(msg.get("subscriptionName"), instanceOf(String.class));
+        assertThat(callback.getMessage().getHeaders().get("X-Request-Id"), instanceOf(String.class));
```

Comment:
This way we don't get an NPE/cast error if the values are wrong. The one test that uses these values now assigns them after the asserts.